raft.go

```diff
@@ -793,7 +901,7 @@ func (r *raft) becomeLeader() {
 	// uncommitted log quota. This is because we want to preserve the
 	// behavior of allowing one entry larger than quota if the current
 	// usage is zero.
-	r.reduceUncommittedSize([]pb.Entry{emptyEnt})
+	r.uncommittedSize = 0
```

Comment:
Circling back on https://github.com/etcd-io/etcd/pull/14627#discussion_r1021299292

Isn't this line unnecessary in the first place? The empty entry is zero-sized and we did call `r.reset()`. Let's add a unit test instead verifying that `payloadsSize(emptyEntry) == 0` and be done with it.

I filed https://github.com/etcd-io/raft/issues/11 for tracking uncommitted size across leadership changes.

Reply:
Thanks for opening the issue, and good point about this being unnecessary.

I switched this back to `r.reduceUncommittedSize(payloadSize(emptyEnt))` for now to make fewer assumptions. I also opened https://github.com/etcd-io/raft/issues/12 to remove the line entirely. Mind slapping `good first issue` and `help wanted` labels on it?